File: Solutions For Interview Prep Kit/Stacks and Queues/largest_rectangle.cpp
long largestRectangle(vector<int> h) {
long mx = 0;
for(int i = 0; i < h.size(); i++){
    long lmn = 0;
    long rmn = 0;
    for(int j = i-1; j >= 0; j--){
        if(h[i] <= h[j]){
            lmn++;
        }
        else{
            break;
        }
    }
    for(int j = i+1; j < h.size(); j++){
        if(h[i] <= h[j]){
            rmn++;
        }
        else{
            break;
        }
    }
    long area = h[i]*(lmn+rmn+1);
    mx = max(mx,area);
}
return mx;
}

File: Solutions For Interview Prep Kit/Stacks and Queues/largest_rectangle.py
def largestRectangle(h):
    #var for storing maximum area
    mx = 0
    for i in range(len(h)):
        # for storing indexes of left and right minimum
        # element which the next most to the present element
        lmn = 0
        rmn = 0

        for j in range(i-1,-1,-1):
            if h[i] <= h[j]:
                lmn += 1
            else:
                break
        for j in range(i+1,len(h)):
            if h[i] <= h[j]:
                rmn += 1
            else:
                break
        # calculating area of the histogram formed
        area = h[i]*(lmn+rmn+1)
        mx = max(mx,area)

    return mx
